sorc/global_cycle.fd/read_write_data.f90

```diff
@@ -1499,4 +1501,352 @@ SUBROUTINE READ_DATA(TSFFCS,SMCFCS,SNOFCS,STCFCS, &
+  ni=kgds(2)                ! 720 for 0.5 x 0.5
+  nj=kgds(3)                ! 360 for 0.5 x 0.5 resolution
+
+  mscan=kgds(11)
+  kb1=ibits(mscan,7,1)   ! i scan direction
+  kb2=ibits(mscan,6,1)   ! j scan direction
+  kb3=ibits(mscan,5,1)   ! (i,j) or (j,i)
+
+! get i and j scanning directions from kb1 and kb2.
+! 0 yields +1, 1 yields -1. +1 is west to east, -1 is east to west.
+  iincdir = 1-kb1*2
+
+! 0 yields -1, 1 yields +1. +1 is south to north, -1 is north to south.
+  jincdir = kb2*2 - 1
+
+! write(*,*) 'read_tf_clim_grb,iincdir,jincdir : ',iincdir,jincdir
+  do k=1,kf
+
+!    kb3 from scan mode indicates if i points are consecutive
+!    or if j points are consecutive
+     if(kb3==0)then     !  (i,j)
+        i=(ni+1)*kb1+(mod(k-1,ni)+1)*iincdir
+        j=(nj+1)*(1-kb2)+jincdir*((k-1)/ni+1)
+     else                !  (j,i)
+        j=(nj+1)*(1-kb2)+(mod(k-1,nj)+1)*jincdir
+        i=(ni+1)*kb1+iincdir*((k-1)/nj+1)
+     endif
+     sst (i,j) = f(k)
+  end do
+     
+  deallocate(lb,f)
+
+  call baclose(lu_sst,iret)
+  if (iret /= 0 ) then
+     write(6,*)'read_tf_clm_grb:  ***error*** close sst file'
+     stop
+  endif
+  
+end subroutine read_tf_clim_grb
+
+subroutine get_tf_clm_dim(file_sst,mlat_sst,mlon_sst)
+!                .      .    .                                       .
+! abstract:   get_tf_clm_dim :  get dimension of rtg sst climatology
+!   prgmmr: xu li            org: np23                date: 2019-03-13
+!
+!
+! program history log:
+!
+!   input argument list:
+!     file_sst - file name of grib sst file
+! output
+!     mlat_sst,mlon_sst
+!
+!     call subs: getgbh
+!
+! attributes:
+!   language: f90
+!   machine:  ibm rs/6000 sp
+!
+!$$$
+  implicit none
+
+! declare passed variables and arrays
+  character(*)                                   , intent(in ) :: file_sst
+  integer                                        , intent(out) :: mlat_sst,mlon_sst
+
+! declare local parameters
+  integer,parameter:: lu_sst = 21   ! fortran unit number of grib sst file
+
+  integer :: iret
+  integer :: mscan,kb1
+  integer :: kf,kg,k,j
+  integer, dimension(22):: jgds,kgds
+  integer, dimension(25):: jpds,kpds
+
+!************+******************************************************************************
+!
+! open sst analysis file (grib)
+  call baopenr(lu_sst,trim(file_sst),iret)
+  if (iret /= 0 ) then
+     write(6,*)'get_tf_clm_dim:  ***error*** opening sst file'
+     stop
+  endif
+
+! define sst variables for read
+  j=-1
+  jpds=-1
+  jgds=-1
+  jpds(5)=11        ! sst variable
+  jpds(6)=1         ! surface
+  jpds(9) = 1
+  call getgbh(lu_sst,0,j,jpds,jgds,kg,kf,k,kpds,kgds,iret)
+
+  mlat_sst = kgds(3)   ! number points on longitude circle (360)
+  mlon_sst = kgds(2)   ! number points on latitude circle (720)
+
+  write(*,*) 'mlat_sst, mlon_sst : ',mlat_sst, mlon_sst
+
+  call baclose(lu_sst,iret)
+  if (iret /= 0 ) then
+     write(6,*)'get_tf_clm_dim:  ***error*** close sst file'
+     stop
```

Comment:
I would use 'mpi_abort' instead of 'stop'.

Reply:
OK, I will use mpi_abort instead of stop